packages/framework/src/Component/Domain/Domain.php

```diff
@@ -20,21 +20,14 @@ class Domain implements DomainIdsProviderInterface
 
     protected ?DomainConfig $currentDomainConfig = null;
 
-    /**
-     * @var \Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig[]
-     */
-    protected array $domainConfigs;
-
-    protected Setting $setting;
-
     /**
      * @param \Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig[] $domainConfigs
      * @param \Shopsys\FrameworkBundle\Component\Setting\Setting $setting
      */
-    public function __construct(array $domainConfigs, Setting $setting)
-    {
-        $this->domainConfigs = $domainConfigs;
-        $this->setting = $setting;
+    public function __construct(
+        protected array $domainConfigs,
```

Comment:
I am missing context

Reply:
oh, sorry :) 
The `$domainConfigs` seems like it may be `readonly`, but it's impossible due to proxy implementation (friendsofphp/proxy-manager-lts). 

when the domain service is not `lazy` in `services.yaml`, it's not wrapped by proxy interface, therefore it may be readonly. 

But the current state is ok :)